pkg/render/render.go

```diff
@@ -159,25 +169,25 @@ func OAuthAppEnv(ctx context.Context, db kclient.Client, oauthAppNames []string,
 	return extraEnv, nil
 }
 
-func addKnowledgeTools(ctx context.Context, db kclient.Client, agent *v1.Agent, thread *v1.Thread, extraEnv []string) ([]string, error) {
+func addKnowledgeTools(ctx context.Context, db kclient.Client, agent *v1.Agent, thread *v1.Thread, extraEnv []string) ([]string, bool, error) {
```

Comment:
This function seems improperly named. It sounds like it doesn't actually add the knowledge tools, and instead just gives you an env var, and returns whether or not the knowledge tool is actually needed. I think it should be renamed to something more descriptive.

Reply:
sound good. I will take the chance to rename